### PR TITLE
Move PHPUnit to GitHub Actions

### PR DESCRIPTION
**Pull request by LukasReschke**
Ref https://github.com/nextcloud/server/pull/27352#issuecomment-853340050

0 clue if this works. We'll see...

### CODE REVIEW

File: .github/workflows/phpunit.yml [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+          --health-timeout 5s
+          --health-retries 5
+        ports:
+          - "6379:6379"
+
+  phpunit-sqlite:
+    runs-on: ubuntu-20.04
+    strategy:
+      matrix:
+        php-versions: [ '7.3', '7.4', '8.0' ]
```
Comment (skjnldsv) [inline .github/workflows/phpunit.yml:52]: Maybe we should do a global ENV as well?

**Comment by skjnldsv** [comment]
```
There was 1 error:

1) Test\Memcache\MemcachedTest::testClear
Exception: Error 47 interacting with memcached : SERVER HAS FAILED AND IS DISABLED UNTIL TIMED RETRY in /home/runner/work/server/server/lib/private/Memcache/Memcached.php:226
Stack trace:
#0 /home/runner/work/server/server/lib/private/Memcache/Memcached.php(118): OC\Memcache\Memcached->verifyReturnCode()
#1 /home/runner/work/server/server/tests/lib/Memcache/MemcachedTest.php(20): OC\Memcache\Memcached->set('PwW69UeB7thlK', '70h9ZAajXh3Ld')
#2 phar:///usr/local/bin/phpunit/phpunit/Framework/TestSuite.php(468): Test\Memcache\MemcachedTest::setUpBeforeClass()
#3 phar:///usr/local/bin/phpunit/phpunit/Framework/TestSuite.php(508): PHPUnit\Framework\TestSuite->run(Object(PHPUnit\Framework\TestResult))
#4 phar:///usr/local/bin/phpunit/phpunit/Framework/TestSuite.php(508): PHPUnit\Framework\TestSuite->run(Object(PHPUnit\Framework\TestResult))
#5 phar:///usr/local/bin/phpunit/phpunit/TextUI/TestRunner.php(487): PHPUnit\Framework\TestSuite->run(Object(PHPUnit\Framework\TestResult))
#6 phar:///usr/local/bin/phpunit/phpunit/TextUI/Command.php(124): PHPUnit\TextUI\TestRunner->run(Object(PHPUnit\Framework\TestSuite), Array, Array, true)
#7 phar:///usr/local/bin/phpunit/phpunit/TextUI/Command.php(93): PHPUnit\TextUI\Command->run(Array, true)
#8 /usr/local/bin/phpunit(1119): PHPUnit\TextUI\Command::main()
#9 {main}
--
```

**Comment by skjnldsv** [comment]
> ```
> There was 1 error:
> 
> 1) Test\Memcache\MemcachedTest::testClear
> Exception: Error 47 interacting with memcached : SERVER HAS FAILED AND IS DISABLED UNTIL TIMED RETRY in /home/runner/work/server/server/lib/private/Memcache/Memcached.php:226
> Stack trace:
> #0 /home/runner/work/server/server/lib/private/Memcache/Memcached.php(118): OC\Memcache\Memcached->verifyReturnCode()
> #1 /home/runner/work/server/server/tests/lib/Memcache/MemcachedTest.php(20): OC\Memcache\Memcached->set('PwW69UeB7thlK', '70h9ZAajXh3Ld')
> #2 phar:///usr/local/bin/phpunit/phpunit/Framework/TestSuite.php(468): Test\Memcache\MemcachedTest::setUpBeforeClass()
> #3 phar:///usr/local/bin/phpunit/phpunit/Framework/TestSuite.php(508): PHPUnit\Framework\TestSuite->run(Object(PHPUnit\Framework\TestResult))
> #4 phar:///usr/local/bin/phpunit/phpunit/Framework/TestSuite.php(508): PHPUnit\Framework\TestSuite->run(Object(PHPUnit\Framework\TestResult))
> #5 phar:///usr/local/bin/phpunit/phpunit/TextUI/TestRunner.php(487): PHPUnit\Framework\TestSuite->run(Object(PHPUnit\Framework\TestResult))
> #6 phar:///usr/local/bin/phpunit/phpunit/TextUI/Command.php(124): PHPUnit\TextUI\TestRunner->run(Object(PHPUnit\Framework\TestSuite), Array, Array, true)
> #7 phar:///usr/local/bin/phpunit/phpunit/TextUI/Command.php(93): PHPUnit\TextUI\Command->run(Array, true)
> #8 /usr/local/bin/phpunit(1119): PHPUnit\TextUI\Command::main()
> #9 {main}
> --
> ```

I dunno, @LukasReschke I'll let you take over :shrug: :thinking:

**Comment by LukasReschke** [comment]
Added memcached as image and now the memcached tests fail 🙈 

```
There was 1 failure:
1527

1528
1) Test\Memcache\MemcachedTest::testClear
1529
Failed asserting that true is false.
1530

1531
/home/runner/work/server/server/tests/lib/Memcache/MemcachedTest.php:40
1532
```

**Comment by LukasReschke** [comment]
The SQLite tests fail with this:

```

2148
There were 6 failures:
2149

2150
1) Test\Route\RouterTest::testGenerateConsecutively
2151
Failed asserting that two strings are equal.
2152
--- Expected
2153
+++ Actual
2154
@@ @@
2155
-'/index.php/apps/files/'
2156
+''
2157

2158
/home/runner/work/server/server/tests/lib/Route/RouterTest.php:41
2159

2160
2) OCA\Files_Sharing\Tests\ApiTest::testCreateShareLink
2161
Failed asserting that two strings are equal.
2162
--- Expected
2163
+++ Actual
2164
@@ @@
2165
-'http://localhost/index.php/s/NpfcdaDrJjt5Kws'
2166
+'http://localhost/'
2167

2168
/home/runner/work/server/server/apps/files_sharing/tests/ApiTest.php:223
2169

2170
3) OCA\Files_Sharing\Tests\ApiTest::testCreateShareLinkPublicUpload
2171
Failed asserting that two strings are equal.
2172
--- Expected
2173
+++ Actual
2174
@@ @@
2175
-'http://localhost/index.php/s/TocCzrgpxxKXHof'
2176
+'http://localhost/'
2177

2178
/home/runner/work/server/server/apps/files_sharing/tests/ApiTest.php:251
2179

2180
4) OCA\Files_Sharing\Tests\ApiTest::testPublicLinkUrl
2181
Failed asserting that two strings are equal.
2182
--- Expected
2183
+++ Actual
2184
@@ @@
2185
-'http://localhost/index.php/s/R8P5K9ndnG4jJ7S'
2186
+'http://localhost/'
2187

2188
/home/runner/work/server/server/apps/files_sharing/tests/ApiTest.php:435
2189

2190
5) OCA\Files_Sharing\Tests\ApiTest::testPublicLinkExpireDate with data set #0 ('2021-06-08', true)
2191
Failed asserting that two strings are equal.
2192
--- Expected
2193
+++ Actual
2194
@@ @@
2195
-'http://localhost/index.php/s/SgT7Hf28YF46BXs'
2196
+'http://localhost/'
2197

2198
/home/runner/work/server/server/apps/files_sharing/tests/ApiTest.php:1301
2199

2200
6) OCA\Files_Sharing\Tests\ApiTest::testCreatePublicLinkExpireDateValid
2201
Failed asserting that two strings are equal.
2202
--- Expected
2203
+++ Actual
2204
@@ @@
2205
-'http://localhost/index.php/s/sgkTDLpfiTkZt6o'
2206
+'http://localhost/'
2207

2208
/home/runner/work/server/server/apps/files_sharing/tests/ApiTest.php:1330
2209
```

**Comment by skjnldsv** [comment]
Any news? :)

**Comment by LukasReschke** [comment]
Nope. And currently no capacity for this. Sorry :(

**Pull request closed without merging**